core/include/kernel/interrupt.h

```diff
@@ -287,11 +319,137 @@ TEE_Result interrupt_alloc_add_handler(struct itr_chip *chip, size_t it_num,
+ *		TEE_SUCCESS in case of success
+ *		TEE_ERROR_DEFER_DRIVER_INIT if clock is not initialized
+ *		Any TEE_Result compliant code in case of error.
+ *
+ * Returns an allocated struct itr_line pointer referencing to an interrupt
+ * line or NULL if invalid description in which case @res provides the error
+ * code. The interrupt matches the reference found with properties of the
+ * pointer FDT node.
+ *
+ * Upon success, the interrupt is configured and consumer can add a handler
+ * function to the interrupt. Yet, the interrupt is not enabled until consumer
+ * calls interrupt_enable().
+ * Upon success, struct itr_line must point to memory allocated with malloc()
+ * or like as it is freed prior returning to caller function.
+ */
+typedef struct itr_line *(*dt_get_itr_func)(struct dt_pargs *args, void *data,
+					    TEE_Result *res);
+
+#ifdef CFG_DT
+/**
+ * dt_register_interrupt_provider - Register an interrupt provider
+ *
+ * @fdt		Device tree to work on
+ * @node	Node offset of the interrupt controller in the DT
+ * @dt_get_itr	Callback to match the devicetree interrupt reference with
+ * @data	Data which will be passed to the get_dt_its callback
+ */
+TEE_Result dt_register_interrupt_provider(const void *fdt, int node,
+					  dt_get_itr_func dt_get_itr,
+					  void *data);
+
+/**
+ * dt_get_interrupt_by_index() - Get an interrupt from DT by interrupt index
+ *
+ * @fdt		Device tree to work on
+ * @node	Node offset of the subnode containing interrupt(s) references
+ * @index	Index in "interrupts" or "exteneded-interrupts" property list
+ * @chip	Output interrupt controller clock reference upon success
+ * @itr_num	Output interrupt line upon success
+ *
+ * Return TEE_SUCCESS in case of success
+ * Return TEE_ERROR_DEFER_DRIVER_INIT if interrupt controller is not yet inited
+ * Return TEE_ERROR_ITEM_NOT_FOUND if the DT does not reference target interrupt
+ * Return any other TEE_Result compliant code in case of error
+ */
+TEE_Result dt_get_interrupt_by_index(const void *fdt, int node,
+				     unsigned int index, struct itr_chip **chip,
+				     size_t *itr_num);
+
+/**
+ * dt_get_interrupt_by_name() - Get an interrupt from DT by interrupt name
+ *
+ * @fdt		Device tree to work on
+ * @node	Node offset of the subnode containing interrupt(s) references
+ * @name	Name identifier used in "interrupt-names" property
+ * @chip	Output interrupt controller clock reference upon success
+ * @itr_num	Output interrupt line upon success
+ *
+ * Return TEE_SUCCESS in case of success
+ * Return TEE_ERROR_DEFER_DRIVER_INIT if interrupt controller is not yet inited
+ * Return TEE_ERROR_ITEM_NOT_FOUND if the DT does not reference target interrupt
+ * Return any other TEE_Result compliant code in case of error
+ */
+TEE_Result dt_get_interrupt_by_name(const void *fdt, int node, const char *name,
+				    struct itr_chip **chip, size_t *itr_num);
+#else
+static inline
+TEE_Result dt_register_interrupt_provider(const void *dt __unused,
```

Comment:
Both `clang-format` and I prefer breaking the line as:
``` C
static inline TEE_Result
dt_register_interrupt_provider(...
```
It's also as far as I know the way it's normally done in the kernel when needed.

Reply:
ok, i'll update.